### PR TITLE
Feature: Add error logger.

### PR DESCRIPTION
**Pull request by driv3r**
Props to Honeybadger, as this work was heavily based on their Logger. Basically it sends error reports from OTP/tasks/processes/etc. So you don't need to manually wrap every expression within try/rescue block.

### CODE REVIEW

**Comment by jarednorman** [comment]
Ooh. Cool.

File: config/config.exs [hunk 1]
```diff
@@ -1,4 +1,2 @@
 use Mix.Config
 
-config :bugsnag,
```
Comment (mootpointer) [inline config/config.exs]: Was this intentionally included? I feel like it's part of an unrelated change.

**Comment by mootpointer** [comment]
Apart from the above, this looks great. I'd like to work towards being able to have reporters which are injected, so that we can avoid the HTTP mocking for testing error reporting, but that's out of scope for this change.

File: config/config.exs (hunk 1 above)
Comment (driv3r) [inline config/config.exs]: yes, it was intentionally, as I've moved configuration to the app module instead

**Comment by driv3r** [comment]
@mootpointer good point :+1: I could do it as well in the next iteration

File: config/config.exs (hunk 1 above)
Comment (mootpointer) [inline config/config.exs]: Ah. Rereading I see it now. It'd be nice to separate those changes, just so their intentions are clear. I might pull the config change into a separate commit which you could rebase these commits on top of to get a clean and clear history. How does that sound?

Comment (driv3r) [inline config/config.exs]: it"s already in a separate commit, shouldn't this be enough?

Comment (mootpointer) [inline config/config.exs]: Yeah, I guess it is – sorry it's been a long day and I was just looking at the aggregated changes view.

Comment (driv3r) [inline config/config.exs]: No problem at all :)

**Comment by mootpointer** [comment]
Having run the tests, my only request is that we suppress the errors and logging during the tests using `ExUnit.CaptureIO` and `ExUnit.CaptureLog`.

**Comment by driv3r** [comment]
@mootpointer added capture log in logger tests, also found one issue with existing bugsnag payload, but I will prepare a separate PR
